File: code/java/decodeString.java
class Solution {
    public String decodeString(String s) {
        String res = "";
        Stack<Integer> stack1 = new Stack<Integer>();
        Stack<String> stack2 = new Stack<String>();
        int number = 0;
        String string = "";
        int i =  0;
        while (i < s.length()){
            if(s.charAt(i)>='0' && s.charAt(i) <= '9'){
                number = 10*number + Character.getNumericValue(s.charAt(i)) ;        
            }
            else if(s.charAt(i) == '['){
                stack2.push("[");
                stack1.push(number);
                number = 0;
            }
            else if(s.charAt(i) == ']'){
                String temp = stack2.pop();
                while(!temp.equals("[")){
                    string = temp + string;
                    temp = stack2.pop();
                }
                int num = stack1.pop();
                temp = "";
                for(int q=0;q<num;q++)
                    temp = temp + string;
                stack2.push(temp);
                string = "";
            }
            else
                stack2.push(s.substring(i,i+1));
                      
            i = i + 1;
        }
        if(!stack2.isEmpty()){
            while(!stack2.isEmpty()){
                String rem = (String)stack2.pop();
                if(!rem.equals("["))
                    res = rem + res;
                    
            }
        }
        return res;
    }    
    
}

File: code/python/decodeString.py
class Solution:
    def decodeString(self, s: str) -> str:
        res = ""
        stack1 = []
        stack2 = []
        number = 0
        string = ""
        i =  0
        while i < len(s):
            if(s[i]>='0' and s[i] <= '9'):
                number = 10*number + int(s[i])              
            elif(s[i] == '['):
                stack2.append('[')
                stack1.append(number)
                number = 0
            elif(s[i] == ']'): 
                while stack2[-1] != '[':
                    string = stack2.pop(-1) + string
                temp = stack1.pop(-1) * string
                stack2[-1] = temp
                string = ""
                
            else:
                stack2.append(s[i])
            i = i + 1
        
        if len(stack2) != 0:
            for i in range(len(stack2)):
                if stack2[i] != '[':
                    res = res + stack2[i]
        return res
                
                
